### ISSUE TITLE
DTZ007 False positive

### ISSUE THREAD

**Issue by ThirVondukr** [charliermarsh/ruff#1296]
Ruff Version: 0.0.188

Ruff correctly detects an error in this code and suggests adding `.replace`:
```py
from datetime import datetime


def parse_datetime(date: str) -> datetime:
    date_format = "%d/%m/%Y %H:%M"
    return datetime.strptime(date, date_format)
```
```
test.py:6:12: DTZ007 The use of `datetime.datetime.strptime()` without %z must be followed by `.replace(tzinfo=)`
```

However ruff reports the same error after `.replace` is added:
```py
from datetime import datetime, timezone


def parse_datetime(date: str) -> datetime:
    date_format = "%d/%m/%Y %H:%M"
    return datetime.strptime(date, date_format).replace(tzinfo=timezone.utc)
```
```
test.py:6:12: DTZ007 The use of `datetime.datetime.strptime()` without %z must be followed by `.replace(tzinfo=)`
```

**Comment by charliermarsh** [comment]
Thank you! Will fix today.

**Comment by charliermarsh** [comment]
Going out in [v0.0.189](https://github.com/charliermarsh/ruff/releases/tag/v0.0.189).
